src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -194,52 +194,95 @@ public void importEntries(List<BibEntry> entries) {
-        BibEntry cleanedEntry = cleanup.doPostCleanup(entry);
-        BibEntry entryToInsert = cleanedEntry;
+        BibEntry cleanedEntry = cleanUpEntry(bibDatabaseContext, entry);
+        Optional<BibEntry> existingDuplicateInLibrary = findDuplicate(bibDatabaseContext, cleanedEntry);
 
-        Optional<BibEntry> existingDuplicateInLibrary = new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(bibDatabaseContext.getDatabase(), entryToInsert, bibDatabaseContext.getMode());
+        Optional<BibEntry> entryToInsert = Optional.of(cleanedEntry);
         if (existingDuplicateInLibrary.isPresent()) {
-            DuplicateResolverDialog dialog = new DuplicateResolverDialog(existingDuplicateInLibrary.get(), entryToInsert, DuplicateResolverDialog.DuplicateResolverType.IMPORT_CHECK, bibDatabaseContext, stateManager, dialogService, preferencesService);
-            switch (dialogService.showCustomDialogAndWait(dialog).orElse(DuplicateResolverDialog.DuplicateResolverResult.BREAK)) {
-                case KEEP_RIGHT:
-                    bibDatabaseContext.getDatabase().removeEntry(existingDuplicateInLibrary.get());
-                    break;
-                case KEEP_BOTH:
-                    break;
-                case KEEP_MERGE:
-                    bibDatabaseContext.getDatabase().removeEntry(existingDuplicateInLibrary.get());
-                    entryToInsert = dialog.getMergedEntry();
-                    break;
-                case KEEP_LEFT:
-                case AUTOREMOVE_EXACT:
-                case BREAK:
-                default:
-                   return;
+            BibEntry duplicateHandledEntry = handleDuplicates(bibDatabaseContext, cleanedEntry, existingDuplicateInLibrary.get());
+            entryToInsert = Optional.ofNullable(duplicateHandledEntry);
+            // If handleDuplicates returns null, stop processing
+            if (entryToInsert.isEmpty()) {
+                return;
             }
         }
-        // Regenerate CiteKey of imported BibEntry
+        regenerateCiteKey(entryToInsert.get());
+        bibDatabaseContext.getDatabase().insertEntry(entryToInsert.get());
+
+        setAutomaticFieldsForEntry(entryToInsert.get());
+
+        addEntryToGroups(entryToInsert.get());
+
+        downloadLinkedFiles(entryToInsert.get());
+    }
+
+    public BibEntry cleanUpEntry(BibDatabaseContext bibDatabaseContext, BibEntry entry) {
+        ImportCleanup cleanup = new ImportCleanup(bibDatabaseContext.getMode());
+        return cleanup.doPostCleanup(entry);
+    }
+
+    public Optional<BibEntry> findDuplicate(BibDatabaseContext bibDatabaseContext, BibEntry entryToCheck) {
+        return new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(bibDatabaseContext.getDatabase(), entryToCheck, bibDatabaseContext.getMode());
+    }
+
+    public BibEntry handleDuplicates(BibDatabaseContext bibDatabaseContext, BibEntry originalEntry, BibEntry duplicateEntry) {
```

Comment:
```suggestion
    public Optional<BibEntry> handleDuplicates(BibDatabaseContext bibDatabaseContext, BibEntry originalEntry, BibEntry duplicateEntry) {
```

Reason for rewrite: NEVER return `null` (in modern code)